crypto/bn/bn_gf2m.c

```diff
@@ -1157,12 +1168,13 @@ int BN_GF2m_poly2arr(const BIGNUM *a, int p[], int max)
         }
     }
 
-    if (k < max) {
+    if (k > 0 && p[0] > OPENSSL_ECC_MAX_FIELD_BITS)
```

Comment:
@vdukhovni you need to be careful when backporting this to 1.1.1 and 1.0.2 branches,
not to break the no-ec configuration, which did at that time not implicitly add no-ec2m:
with 1.1.1 the header file can be included, but does not define anything, so
`OPENSSL_ECC_MAX_FILED_BITS` may be undefined here, and with 1.0.2 the
include `openssl/ec.h` above is including a non-existing file.

Reply:
Thanks, the backports are already done, and tested with `noec`, all good.  In 1.0.2 I updated Configure so that `noec` implies `noec2m`. As for `ec.h` it is in 1.0.2, a symlink to `crypto/ec/ec.h` is added to `include/openssl/` as part of the build.